src/main.cpp

```diff
@@ -5653,7 +5653,7 @@ bool SendMessages(CNode* pto, bool fSendTrickle)
         //
         // Message: getdata (non-blocks)
         //
-        while (!pto->fDisconnect && !pto->mapAskFor.empty() && (*pto->mapAskFor.begin()).first <= nNow)
+        while (!pto->fDisconnect && !pto->mapAskFor.empty() && (*pto->mapAskFor.begin()).first <= nNow && !IsInitialBlockDownload())
```

Comment:
made the change and pushed.

On 03/12/2015 6:19 AM, Wladimir J. van der Laan wrote:
>
> In src/main.cpp
> <https://github.com/bitcoin/bitcoin/pull/7164#discussion_r46555484>:
>
> > @@ -5653,7 +5653,7 @@ bool SendMessages(CNode* pto, bool fSendTrickle)
> >          //
> >          // Message: getdata (non-blocks)
> >          //
> > -        while (!pto->fDisconnect && !pto->mapAskFor.empty() && (*pto->mapAskFor.begin()).first <= nNow)
> > +        while (!pto->fDisconnect && !pto->mapAskFor.empty() && (*pto->mapAskFor.begin()).first <= nNow && !IsInitialBlockDownload())
>
> No need to check IsInitialBlockDownload() every iteration of the while
> loop.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/7164/files#r46555484>.
>